### DISCUSSION TITLE
ImportError: can't import name Microdot on helloworld sample for RP2040

### DISCUSSION THREAD

**Discussion by pczekalski** [miguelgrinberg/microdot#171] [Q&A] [answered]
**Describe the bug**
I'm trying to run a simple helloworld app on Raspberry Pi Pico W (Micropython)

**To Reproduce**
Upload microdot.py, author app.py:
```
import network
import socket
from time import sleep
import machine
from microdot import Microdot #local library
ssid = '...'
password='...'
def connect():
    #Connect to WLAN
    wlan = network.WLAN(network.STA_IF)
    wlan.active(True)
    wlan.connect(ssid, password)
    while wlan.isconnected() == False:
        print('Waiting for connection...')
        sleep(1)
    print(wlan.ifconfig())
try:
    connect()
except KeyboardInterrupt:
    machine.reset()

```

**Expected behavior**
Code should run, but it does not, fails with

```>>> 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/app.py", line 5, in <module>
ImportError: can't import name Microdot
```

**Additional context**
files seem to be present OK and library does contain Microdot class:
```
>>> import os
>>> os.listdir()
['app.py', 'microdot.py']
```
![Screenshot from 2023-10-01 16-01-09](https://github.com/miguelgrinberg/microdot/assets/23726790/251d070a-374a-4581-ab60-113540c85003)

**Comment by miguelgrinberg** [comment, marked as answer]
Please do not write issues in this project when you need help fixing an issue in your own application. The discussions board is for questions and requesting help. This time I have converted your issue to a discussion for you.

My suggestion is that you erase the flash memory in your device, then install a fresh MicroPython and copies of your source files. Hopefully that will clear any problems that you have.

**Reply by pczekalski** [reply]
Thanks, I acknowledge, cleaning flash [RP2040 Reseting Flash Memory](https://www.raspberrypi.com/documentation/microcontrollers/raspberry-pi-pico.html#resetting-flash-memory) and reflashing latest Micropython solved the problem.
